Zend/zend_ast.c

```diff
@@ -441,3 +444,1011 @@ ZEND_API void zend_ast_apply(zend_ast *ast, zend_ast_apply_func fn) {
+			} else {
+				smart_str_appends(str, "{\n");
+				zend_ast_export_stmt(str, ast->child[1], indent + 1);
+				zend_ast_export_indent(str, indent);
+				smart_str_appendc(str, '}');
+				break;
+			}
+			break;
+		case ZEND_AST_SWITCH:
+			smart_str_appends(str, "switch (");
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			smart_str_appends(str, ") {\n");
+			zend_ast_export_ex(str, ast->child[1], 0, indent + 1);
+			zend_ast_export_indent(str, indent);
+			smart_str_appendc(str, '}');
+			break;
+		case ZEND_AST_SWITCH_CASE:
+			zend_ast_export_indent(str, indent);
+			if (ast->child[0]) {
+				smart_str_appends(str, "case ");
+				zend_ast_export_ex(str, ast->child[0], 0, indent);
+				smart_str_appends(str, ":\n");
+			} else {
+				smart_str_appends(str, "default:\n");
+			}
+			zend_ast_export_stmt(str, ast->child[1], indent + 1);
+			break;
+		case ZEND_AST_DECLARE:
+			smart_str_appends(str, "declare(");
+			if (ast->child[0]->kind == ZEND_AST_CONST_DECL) {
+				zend_ast_export_list(str, (zend_ast_list*)ast->child[0], 1, 0, indent);
+			} else {
+				zend_ast_export_ex(str, ast->child[0], 0, indent);
+			}
+			if (ast->child[1]) {
+				smart_str_appends(str, " = ");
+				zend_ast_export_ex(str, ast->child[1], 0, indent);
+			}
+			smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_PROP_ELEM:
+			smart_str_appendc(str, '$');
+		case ZEND_AST_CONST_ELEM:
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			APPEND_DEFAULT_VALUE(1);
+		case ZEND_AST_USE_TRAIT:
+			smart_str_appends(str, "use ");
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			if (ast->child[1]) {
+				smart_str_appends(str, " {\n");
+				zend_ast_export_ex(str, ast->child[1], 0, indent + 1);
+				zend_ast_export_indent(str, indent);
+				smart_str_appends(str, "}");
+			} else {
+				smart_str_appends(str, ";");
+			}
+			break;
+		case ZEND_AST_TRAIT_PRECEDENCE:
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			smart_str_appends(str, " insteadof ");
+			zend_ast_export_ex(str, ast->child[1], 0, indent);
+			break;
+		case ZEND_AST_METHOD_REFERENCE:
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			smart_str_appends(str, "::");
+			zend_ast_export_name(str, ast->child[1], 0, indent);
+			break;
+		case ZEND_AST_NAMESPACE:
+			smart_str_appends(str, "namespace");
+			if (ast->child[0]) {
+				smart_str_appendc(str, ' ');
+				zend_ast_export_name(str, ast->child[0], 0, indent);
+			}
+			if (ast->child[1]) {
+				smart_str_appends(str, " {\n");
+				zend_ast_export_stmt(str, ast->child[1], indent + 1);
+				zend_ast_export_indent(str, indent);
+				smart_str_appends(str, "}\n");
+			} else {
+				smart_str_appendc(str, ';');
+			}
+			break;
+		case ZEND_AST_USE_ELEM:
+		case ZEND_AST_TRAIT_ALIAS:
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			if (ast->child[1]) {
+				smart_str_appends(str, " as ");
+				zend_ast_export_name(str, ast->child[1], 0, indent);
+			}
```

Comment:
For `TRAIT_ALIAS` we also need to handle the case where only modifiers are changed, e.g. `a as public`.